### DISCUSSION TITLE
How to use dataclasses generated from `import_xml` ?

### DISCUSSION THREAD

**Discussion by rempAut** [FreeOpcUa/opcua-asyncio#1749] [Q&A] [answered]
Hello, Im looking for a proper way to get typehints for my structures.

In opcua-asyncio server, I am importing PLC structures from a nodeset. Everything works as expected, the structures are created in the ua module, but I don't have any typehint info if I do something like this:
```
plcstruct = ua.myImportedPlcStruct(param1,param2)
```

Is there a way to get the typehints? Am I missing something obvious here? Thanks in advance for any help.

**Comment by schroeder-** [comment]
The structs are python dataclasses, you can get all fields via [fields](https://docs.python.org/3/library/dataclasses.html#dataclasses.fields).

**Comment by rempAut** [comment]
Thanks, but that's not exactly what I need. I would just like to have typehints for these classes, something like this:

<img src="https://github.com/user-attachments/assets/ac98daa7-9e80-4d89-b00c-2164d2fce127" width="400">

Is there a way to achieve this? My system has opcua python server, and a PLC client. Ultimately I want to be able to define my "interface" structures one one side, and import them on the other. I was thinking about some other options:

* create the dataclasses on the python server, then use the `new_struct()` method to create the structures on the server. This comes with a disadvantage of maintaining the "PLC" dataclasses instead of simply importing them...

**Comment by rempAut** [comment, marked as answer]
I am closing this, because the title is missleading and not so relevant to what I am really trying to acomplish.
